### ISSUE TITLE
Medline search throws Exception and hangs (Jabref 3.6)

### ISSUE THREAD

**Issue by cookpa** [JabRef/jabref#2110]
JabRef version 3.6 on Mac OS X El Capitan

Steps to reproduce:

1. Open web search
2. Enter "Cook PA" (with or without quotes)
3. It says "Number of references 105. Number to fetch? [click OK]
4. The blue progress bar fills up, but the references never appear

Sometimes I can run a query and some references will appear, but then it hangs. For example, if I search "Tustison [au]" (no quotes), I get some results but then it hangs at about 25% progress. The hangups are accompanied by Exceptions in the console window:

```
Exception in thread "JabRef CachedThreadPool" java.lang.NullPointerException
	at net.sf.jabref.logic.importer.fileformat.MedlineImporter.addArticleInformation(MedlineImporter.java:516)
	at net.sf.jabref.logic.importer.fileformat.MedlineImporter.parseArticle(MedlineImporter.java:341)
	at net.sf.jabref.logic.importer.fileformat.MedlineImporter.importDatabase(MedlineImporter.java:151)
	at net.sf.jabref.gui.importer.fetcher.MedlineFetcher.fetchMedline(MedlineFetcher.java:226)
	at net.sf.jabref.gui.importer.fetcher.MedlineFetcher.processQuery(MedlineFetcher.java:199)
	at net.sf.jabref.gui.importer.fetcher.GeneralFetcher.lambda$actionPerformed$5(GeneralFetcher.java:223)
	at net.sf.jabref.gui.importer.fetcher.GeneralFetcher$$Lambda$191/1683863109.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

IEEE and Springer search don't seem to have this problem. 

I can get specific things from Medline, for example if I enter a single PMID I can get the result. But larger searches sometimes fail with the above error. I had thought it was due to Medline refusing large requests, but I can enter some very general terms (eg searching my last name without initials) and get hundreds of results each time.

I tried the same thing on 3.0 and it works correctly, I can search "Cook PA [au]" and the results show up as expected.

**Comment by tobiasdiez** [comment]
@tschechlovdev could you please have a look as you recently rewrote the MedlineImporter in #1479. Apparently the `issn` is null https://github.com/JabRef/jabref/blame/ee688c44765f11344226da531b63f9df345a6585/src/main/java/net/sf/jabref/logic/importer/fileformat/MedlineImporter.java#L515

**Comment by Siedlerchr** [comment]
It would be interesting to see if the bug is still present in the latest dev version

**Comment by Siedlerchr** [comment]
I just found the error and added a quick fix for it. Works now as expected

**Comment by cookpa** [comment]
Wonderful, thanks.

**Comment by Siedlerchr** [comment]
@cookpa  You can use the latest development from here, it includes the fix.  http://builds.jabref.org/medlineIssn/
If you note further errors, please report/open a new issue (if not the same)

**Comment by cookpa** [comment]
Thanks, I downloaded `JabRef_macos_3_7dev--snapshot--2016-10-03--medlineIssn--dc77ad8.dmg`. 

It solves the original problem but now the search hangs after the first successful search. I searched Medline for "Tustison [au]". It reported 53 results, but the number to retrieve box was blank so I entered 53. It completed successfully and I could import the citations.

Unfortunately, if I try a subsequent search, the program hangs. It displays the number of results dialog but after I click OK, Jabref becomes unresponsive. I have to force quit, so I can't get to the error console.

I've tried this with a few searches. The initial search works fine, and doesn't show the old error. But then I can't search again.
